quicklens/src/main/scala-3/com/softwaremill/quicklens/QuicklensMacros.scala

```diff
@@ -363,30 +341,30 @@ object QuicklensMacros {
         }
       } else if isProduct(objSymbol) || isProductLike(objSymbol) then {
         val argsMap: Map[String, Term] = fields.map { (field, trees) =>
-          val (fieldMethod, name) = field match {
+          val fieldMethod = field match {
             case PathSymbol.Field(name) =>
-              symbolAccessorByNameOrError (obj, name) -> name
+              symbolAccessorByNameOrError(obj, name)
             case PathSymbol.Extension(term, name) =>
-              val extensionMethod = symbolAccessorByNameOrError (term, name)
-              Apply(extensionMethod, List(obj)) -> name
+              val extensionMethod = symbolAccessorByNameOrError(term, name)
+              Apply(extensionMethod, List(obj))
           }
           val resTerm: Term = trees.foldLeft[Term](fieldMethod) { (term, tree) =>
             mapToCopy(owner, mod, term, tree)
           }
-          val namedArg = NamedArg(name, resTerm)
-          name -> namedArg
+          val namedArg = NamedArg(field.name, resTerm)
+          field.name -> namedArg
         }.toMap
         methodSymbolByNameAndArgs(objSymbol, "copy", argsMap) match
           case Right(copy) =>
             callMethod(obj, copy, List(argsMap))
           case Left(error) =>
             val objCompanion = findCompanionLikeObject(objSymbol)
-            objCompanion.flatMap(methodSymbolByNameAndArgs(_, "copy", argsMap).toOption) match
+            methodSymbolByNameAndArgs(objCompanion, "copy", argsMap).toOption match
```

Comment:
Do we really want to call `methodSymbolByNameAndArgs` if/when `objCompanion` is NoSymbol?

Reply:
It will still give an error and this way we don't have two handle two different error channels.